Content.Server/GameObjects/Components/Photography/PhotoCameraComponent.cs

```diff
@@ -0,0 +1,333 @@
+                        if (author.TryGetComponent(out IHandsComponent hands))
+                        {
+                            foreach (var heldItem in hands.GetAllHeldItems())
+                            {
+                                hands.Drop(heldItem.Owner, false);
+                            }
+                        }
+
+                        //Drop their gear
+                        if (author.TryGetComponent(out InventoryComponent inv))
+                        {
+                            inv.UnequipAll(true);
+                        }
+
+                        //Vanish without a trace
+                        author.Delete();
+                    } else
+                    {
+                        if(author.TryGetComponent(out HandsComponent hands)){
+                            hands.PutInHand(photoEnt.GetComponent<ItemComponent>());
+                        }
+                    }
+
+                    //Store photo to disk, assign path to photo entity
+                    _photoSystem.StorePhoto(photoTaken.Data, photo);
+
+                    break;
+            }
+        }
+
+        private void PlaySoundCollection(string name, float volume)
+        {
+            var soundCollection = _prototypeManager.Index<SoundCollectionPrototype>(name);
+            var file = _robustRandom.Pick(soundCollection.PickFiles);
+            EntitySystem.Get<AudioSystem>()
+                .PlayFromEntity(file, Owner, AudioHelpers.WithVariation(0.15f).WithVolume(volume));
+        }
+
+        /// <summary>
+        /// Toggles camera's on/off state
+        /// </summary>
+        private bool ToggleOnOff()
+        {
+            if (CameraOn)
+            {
+                CameraOn = false;
+                _spriteComponent.LayerSetState(1, "camera_off");
+                _audioSystem.PlayFromEntity("/Audio/machines/machine_switch.ogg", Owner, AudioHelpers.WithVariation(0.15f).WithVolume(-5));
+                return true;
+            }
+
+            CameraOn = true;
+            _spriteComponent.LayerSetState(1, "camera_on");
+            _audioSystem.PlayFromEntity("/Audio/machines/machine_switch.ogg", Owner, AudioHelpers.WithVariation(0.15f).WithVolume(-5));
+            return true;
+        }
+
+        public bool UseEntity(UseEntityEventArgs eventArgs)
+        {
+            return ToggleOnOff();
+        }
+
+        public void Examine(FormattedMessage message, bool inDetailsRange)
+        {
+            if (CameraOn)
+            {
+                message.AddMarkup(Loc.GetString("The {0} is [color=green]On[/color]", Owner.Name));
+            }
+            else
+            {
+                message.AddMarkup(Loc.GetString("The {0} is [color=red]Off[/color]", Owner.Name));
+            }
+
+            if (inDetailsRange)
+            {
+                message.AddMarkup(Loc.GetString("\nFilm: [color={0}]{1}/{2}[/color], ",
+                    Film<=0 ? "red" : "white", Film, FilmMax));
```

Comment:
No `HasFilm`?

Reply:
Added `HasFilm` later on, can certainly replace the `FIlm<=0` with it.